charts/redpanda/helpers.go

```diff
@@ -113,11 +113,11 @@ func StatefulSetPodLabels(dot *helmette.Dot, statefulSet map[string]any) map[str
 		statefulSetLabels = values.Statefulset.PodTemplate.Labels
 	}
 
-	defults := map[string]string{
+	defaults := map[string]string{
 		"redpanda.com/poddisruptionbudget": Fullname(dot),
 	}
 
-	return helmette.Merge(statefulSetLabels, StatefulSetPodLabelsSelector(dot, nil), defults)
+	return helmette.Merge(statefulSetLabels, StatefulSetPodLabelsSelector(dot, nil), defaults)
```

Comment:
hmmmm. Could passing in `nil` here potentially cause issues? If this is an upgrade and the existing statefulset had differing labels (IE the first if clause of `StatefulSetPodLabelsSelector` is `true`) then we might be generating incorrect labels here.